src/SpineRuntime/Animation.js

```diff
@@ -0,0 +1,70 @@
+var spine = require('../SpineRuntime') || {};
+spine.Animation = function (name, timelines, duration)
```

Comment:
Well its more than just pattern, this is a circular dependency. Animation.js requires index.js which requires Animation.js. I don't think it is acceptable to leave it like that, you will find many subtle broken things throughout the code with a circular dep like that.